run/action.yaml

```diff
@@ -0,0 +1,128 @@
+name: "Sandard Action Runner"
+description: "Run a Standard action."
+
+inputs:
+  key_path:
+    description: "The path to place the Nix secret key."
+    required: false
+    default: "${{ runner.temp }}/.nix-key"
+  github_pat:
+    description: "GitHub token for accessing private repos."
+    required: false
+    default: ${{ github.token }}
+  nix_key:
+    description: "Secret key used for Nix package signing."
+    required: false
+  s3_key:
+    description: "Key to upload to S3 Buckets with Nix."
+    required: false
+  s3_id:
+    description: "ID to upload to S3 Buckets with Nix."
+    required: false
+  cache:
+    description: "Nix cache to optionally upload artifacts to"
+    # auto refers to the local nix store, so nothing is uploaded by default
+    default: auto
+    required: false
+  builder:
+    description: "Nix store to build on."
+    default: auto
+    required: true
+  json:
+    description: "JSON describing the action, found in the discovery phase."
+    required: true
+  nix_ssh_key:
+    description: "SSH private key to send builds to a remote Nix builder."
+    default: none
+    required: false
+  ssh_known_hosts:
+    description: "SSH known host keys for Nix builders"
+    default: none
+    required: false
+  extra_nix_config:
+    description: "Configuration to append to the nix.conf."
+    required: false
+
```

Comment:
I think for adding additional secrets as required by actions, we just have to.gro this list.

Examples:

- docker hub credentials
- npm / pypi/ rust crate credentials

Reply:
well if GH actions allowed for more structured data as input it'd be nicer. We could just make a simple map of `{ path = secret; }`, maybe we could still do that by inputting a json string?